test/sample_exact_es_percent/processor-spec.ts

```diff
@@ -0,0 +1,206 @@
+import { WorkerTestHarness } from 'teraslice-test-harness';
+import { SampleExactESPercentConfig } from '../../asset/src/sample_exact_es_percent/interfaces.js';
+import { debugLogger, pDelay, TestClientConfig } from '@terascope/job-components';
+
+describe('sample_exact_es_percent', () => {
+    let harness: WorkerTestHarness;
+    const logger = debugLogger('test-logger');
+
+    // Each harness will make a get request to the mock elasticsearch client on initialization, and
+    // once every 'window_ms' the harness exists beyond that. Each request to the client will return
+    // percentArr[i] then increment i for the next request. Update array when modifying tests.
+    const percentArr = [100, 100, 100, 0, 50, 100, 25, 95, 50, 50, 0, 100];
+    let i = 0;
+
+    const defaultClients: TestClientConfig[] = [
+        {
+            type: 'elasticsearch-next',
+            endpoint: 'default',
+            createClient: async () => ({
+                client: {
+                    get: () => {
+                        return { _source: { percent: percentArr[i++] }, found: true };
+                    }
+                },
+                logger
+            }),
+        }
+    ];
+
+    async function makeTest(config: Partial<SampleExactESPercentConfig> = { index: 'my-index', document_id: 'abc123' }, clients = defaultClients) {
+        const baseConfig = {
+            _op: 'sample_exact_es_percent',
+        };
+        const opConfig = Object.assign({}, baseConfig, config);
+        harness = WorkerTestHarness.testProcessor(opConfig, { clients });
+
+        await harness.initialize();
+
+        return harness;
+    }
+
+    afterEach(async () => {
+        if (harness) await harness.shutdown();
+    });
+
+    it('with initial percentage at 100%, should return empty array from empty array', async () => {
```

Comment:
It's pretty confusing where the percentage comes from.  Does it come from `percentArr` somehow which means the order of execution of the tests matters?  If I came along and added a test without updating `percentArr`, everything would break?

I think it would be more straight forward if the percentage were passed in during the`it`.  Though maybe I am missing something.

Reply:
Agreed. I updated the tests so each `it` overrides `percentArr` with a new array of the percentages it expects.